pandas/core/algorithms.py

```diff
@@ -405,7 +406,21 @@ def isin(comps, values):
         values = construct_1d_object_array_from_listlike(list(values))
 
     comps, dtype, _ = _ensure_data(comps)
-    values, _, _ = _ensure_data(values, dtype=dtype)
+    # Convert `values` to `dtype` if dtype of `values` is like `dtype`
```

Comment:
you don't need to check all these conditions, rather you can just do

In [14]: l = [1.0, 1]

```
In [15]: set([type(v) for v in l])
Out[15]: {float, int}
```
If you have a set of len(1) then you can go ahead and pass the dtype (else you can use ``object``).

Reply:
@jreback thx for the comment, not sure if I got you right though. Imagine I have
```python
pd.Series([1, 0]).isin([1.0, 0.5])
```
Then the len of the set would be one (I only have floats), but the conversion to `int` would still not be desirable.

Reply:
thats a different case which is actually simple. the dtype that ensure_data returns is different from the dtype that the set check returns. 

NONE of this checking can be done here, rather it is *already* done in ensure_data